src/java/org/apache/cassandra/serializers/ListSerializer.java

```diff
@@ -104,6 +106,8 @@ public <V> List<T> deserializeForNativeProtocol(V input, ValueAccessor<V> access
             {
                 // We can have nulls in lists that are used for IN values
                 V databb = readValue(input, accessor, offset, version);
+                if (databb == null)
+                    throw new MarshalException("Not enough bytes to read value in list");
                 offset += sizeOfValue(databb, accessor, version);
                 if (databb != null)
```

Comment:
nit: This check has become unnecessary?

Reply:
Why not to throw that exception from that method? We do not need to throw it explicitly everywhere then.

Reply:
I think need to leave the possibility of a null open for lists.

Reply:
For sets and maps, I would almost have a separate `readNonNullValue()` that does throw internally.

Reply:
Went with your suggestion - null in the input seems more likely than corruption so a better, cleaner exception.